torchtune/models/clip/_text_encoder.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+from torch import nn, Tensor
+
+from torchtune.modules import (
+    FeedForward,
+    MultiHeadAttention,
+    TransformerSelfAttentionLayer,
+)
+from torchtune.modules.activations import QuickGELU
+
+
+class CLIPTextEncoder(nn.Module):
+    """
+    Text encoder for CLIP.
+
+    Args:
+        vocab_size (int): size of the vocabulary, default 49408
+        max_seq_len (int): context size, default 77
+        embed_dim (int): embedding/model dimension size, default 768
+        num_heads (int): number of attention heads, default 12
+        num_layers (int): number of transformer layers, default 12
+        norm_eps (float): small value added to denominator for numerical stability, default 1e-5
+    """
+
+    def __init__(
+        self,
+        vocab_size: int = 49408,
+        max_seq_len: int = 77,
+        embed_dim: int = 768,
+        num_heads: int = 12,
+        num_layers: int = 12,
+        norm_eps: float = 1e-5,
+    ):
+        super().__init__()
+        self.max_seq_len = max_seq_len
+
+        self.token_embedding = nn.Embedding(vocab_size, embed_dim)
+        self.position_embedding = nn.Parameter(torch.empty(max_seq_len, embed_dim))
+
+        self.encoder = nn.Sequential(
```

Comment:
For consistency I think this could be structured more like the [Llama3VisionProjectionHead](https://github.com/pytorch/torchtune/blob/e1caa9f82fea24d728f9b244a9dd1957f5ed7465/torchtune/models/llama3_2_vision/_encoder.py#L15) code with a module that takes in "layers,  norm, and embeddings" and then define the text model as a function in component builders. It would make it easier to support a LoRA version of the model.